runtime/runtime/src/state_viewer.rs

```diff
@@ -126,14 +126,16 @@ impl TrieViewer {
             0,
             &empty_hash,
             &view_state.epoch_id,
-            &view_state.last_block_hash,
+            &view_state.block_hash,
             epoch_info_provider,
             view_state.current_protocol_version,
         );
         let config = Arc::new(RuntimeConfig::default());
         let apply_state = ApplyState {
             block_index: view_state.block_height,
-            last_block_hash: view_state.last_block_hash,
+            // Used for legacy reasons
+            last_block_hash: view_state.block_hash,
```

Comment:
I think it is better to add `last_block_hash` to `ViewApplyState` and use it here so that in the future, it we need to actually use `last_block_hash` inside runtime, this would not cause any confusion on the rpc side.

Reply:
I believe we don't expose `ViewApplyState` through RPC, so it is not an issue here. However, the rest of the reasoning on whether to use View* or non-View* structure here might be still relevant (I fail to assess it at the moment).